### ISSUE TITLE
Outdated opentelemetry dependency creating conflicts 

### ISSUE THREAD

**Issue by lewis-anderson53** [sodadata/soda-core#2192]
The currently supported versions of [opentelemetry](https://pypi.org/project/opentelemetry-api/#history) are up to 1.22 which are now a year old. This was last changed in march this year via #2043.

The reason I've stumbled across this is due to [Airflow no longer supporting opentelemetry <1.22](https://github.com/apache/airflow/pull/44113) explicitly in their most recent release.

The package version limit committed to in soda-core in March '24 appears to maintain support for Python 3.7, which makes it more of a tricky discussion I'm sure, however not moving with later versions of other python packages (opentelemetry is just the current problem) will cause many conflicts with other packages in the future I'm sure.

It would be nice to move on from 3.7 which will mean many other packages (e.g markupsafe #2103) can also be bumped.

**Comment by tools-soda** [comment]
CLOUD-9045

**Comment by nathadfield** [comment]
FYI:
```
48.93 ERROR: Cannot install apache-airflow==2.10.4+astro.1 and soda-core-bigquery because these package versions have conflicting dependencies.
48.93 
48.93 The conflict is caused by:
48.93     apache-airflow 2.10.4+astro.1 depends on opentelemetry-api>=1.24.0
48.93     soda-core 3.3.0 depends on opentelemetry-api~=1.16.0
48.93 
```

**Comment by PatziRi** [comment]
I've also found that MarkupSafe has the same issue, as soda-core 3.4.2 depends on MarkupSafe 2.1.2 while airflow requires version 2.1.5

**Comment by ghjklw** [comment]
To add to the discussion, [Python 3.7 reached end-of-life](https://devguide.python.org/versions/#unsupported-versions) in June 2023, a year and a half ago!
There should be no reason to maintain compatibility with a no longer supported version of Python. Those who decide to stick with Python 3.7 should also accept to stick with an older version of Soda, instead of creating dependency conflicts for those who maintain their dependencies updated.

By the way, Python 3.8 is also end-of-life now.

**Comment by swathigottumukkala** [comment]
we are blocked on integrating soda into our existing service due to this issue, appreciate if the work can be 
 prioritized

**Comment by swathigottumukkala** [comment]
may I ask if there are plans to update telemetry dependencies in the near future? If not, I may have time to look into this following the instructions in [CONTRIBUTING.MD](https://github.com/sodadata/soda-core/blob/main/CONTRIBUTING.md).

**Comment by lewis-anderson53** [comment]
The latest release doesn't appear to properly address the issue.

https://github.com/sodadata/soda-core/pull/2199/commits/f80c44397bbdc54e6d6e8b8e7ec72b2c036df62d

This commit sets `"opentelemetry-api~=1.16.0,<2.0.0",` but this still only allows max version to be `1.16.*`.
For example, the airflow conflict `apache-airflow 2.10.4+astro.1 depends on opentelemetry-api>=1.24.0` is still present.

The correct dependency specification I think is `"opentelemetry-api>=1.16.0,<2.0.0",`
I'm unable to re-open the issue due to permissions but I would like to.

**Comment by ghjklw** [comment]
@lewis-anderson53, indeed! Not sure how that happened, that was definitely not my intention. I'll create a new PR later today, and hopefully it won't take as long to merge it.
